src/devices/openxrheadset/OpenXrInterface.cpp

```diff
@@ -1038,6 +1087,84 @@ bool OpenXrInterface::updateInteractionProfiles()
     return true;
 }
 
+void OpenXrInterface::updateHandTracking()
+{
+    // allocate memory for left_hand_joint_locations
+    m_pimpl->right_hand_joint_locations.resize(XR_HAND_JOINT_COUNT_EXT);
+
+    // Allocate memory for left_hand_joint_locations (if not already done)
+    m_pimpl->left_hand_joint_locations.resize(XR_HAND_JOINT_COUNT_EXT);
+
+    m_pimpl->leftHandJointPoses_.clear();
+    m_pimpl->rightHandJointPoses_.clear();
+
+    XrHandJointLocationsEXT left_hand_joints =
+    {
+    .type = XR_TYPE_HAND_JOINT_LOCATIONS_EXT,
+    .next = NULL,
+    .isActive = XR_FALSE,
+    .jointCount = XR_HAND_JOINT_COUNT_EXT,
+    .jointLocations = m_pimpl->left_hand_joint_locations.data()
+    };
+
+    XrHandJointLocationsEXT right_hand_joints =
+    {
+    .type = XR_TYPE_HAND_JOINT_LOCATIONS_EXT,
+    .next = NULL,
+    .isActive = XR_FALSE,
+    .jointCount = XR_HAND_JOINT_COUNT_EXT,
+    .jointLocations = m_pimpl->right_hand_joint_locations.data()
+    };
+
+    const XrHandJointsLocateInfoEXT locate_info =
+    {
+    .type = XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT,
+    .next = NULL,
+    .baseSpace = m_pimpl->play_space,
+    .time = m_pimpl->frame_state.predictedDisplayTime
+    };
+
+    // Locate left hand joints
+    XrResult leftResult = m_pimpl->pfn_xrLocateHandJointsEXT(m_pimpl->left_hand_tracker, &locate_info, &left_hand_joints);
+    if (XR_SUCCEEDED(leftResult) && left_hand_joints.isActive) {
+        for (uint32_t i = 0; i < left_hand_joints.jointCount; ++i) {
+            const XrHandJointLocationEXT& joint = left_hand_joints.jointLocations[i];
+            if (joint.locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
+                Eigen::Vector3f position(joint.pose.position.x, joint.pose.position.y, joint.pose.position.z);
+                Eigen::Quaternionf rotation(joint.pose.orientation.w, joint.pose.orientation.x, joint.pose.orientation.y, joint.pose.orientation.z);
+                OpenXrInterface::Pose handPose;
+                handPose.position = position;
+                handPose.rotation = rotation;
+                handPose.positionValid = true;
+                handPose.rotationValid = true;
+                m_pimpl->leftHandJointPoses_.push_back(handPose);
```

Comment:
Continuing the comment above, if you avoid the clear, remember not to use the ``push_back`` here.